### ISSUE TITLE
[Bug Report]: `ElDatePicker` two way binding bug

### ISSUE THREAD

**Issue by lighter217** [element-plus/element-plus#603]
在组件中选择日期后，无法将选择的值渲染到dom上

**Comment by zazzaz** [comment]
Any demo code please?

**Comment by lighter217** [comment]
而且value-format也不起作用
![11111](https://user-images.githubusercontent.com/44219413/99250029-fdbfec00-2845-11eb-9ec2-ec7a4c6405cb.png)
![22222](https://user-images.githubusercontent.com/44219413/99250040-031d3680-2846-11eb-80ad-1cb438be3c3c.png)

**Comment by denyncrawford** [comment]
Same issue here. I opened an issue about another relatet problem [here](https://github.com/element-plus/element-plus/issues/606) but I also mentioned this error...

**Comment by JeremyWuuuuu** [comment]
I think this might be a bug, but could you add a  watch statement to watch the model see if it's getting chaned?

**Comment by lighter217** [comment]
值确实会变，但是界面不渲染。
![333](https://user-images.githubusercontent.com/44219413/99329443-3816a100-28b9-11eb-934c-787f0e375141.png)
![444](https://user-images.githubusercontent.com/44219413/99329451-3baa2800-28b9-11eb-84f6-f9d85364fbd4.png)

**Comment by zazzaz** [comment]
@lighter217 can you provide a reproduction repo, please. It looks ok on my side.

<img width="382" alt="image" src="https://user-images.githubusercontent.com/10731096/99509738-215b7180-29c1-11eb-8d95-bad665e4486a.png">

```html
<template>
  <div>param: {{ param.start }}</div>
  <el-date-picker
    v-model="param.start"
    type="date"
  />
</template>

<script>
import { reactive } from 'vue'
export default {
  setup() {
    const param = reactive({ starrt: '' })
    return {
      param,
    }
  },
}
</script>

```

**Comment by lighter217** [comment]
still not working.
![777](https://user-images.githubusercontent.com/44219413/99606763-5dd0b100-2a45-11eb-8993-d0764ebb7629.png)
![888](https://user-images.githubusercontent.com/44219413/99606768-5f9a7480-2a45-11eb-81db-ecb433fcdf86.png)

**Comment by JeremyWuuuuu** [comment]
Could you tell us which version of `vue` you were using? We had an investigation in which we believe this issue relates to the the change of `vue` because this repo is depending on version `3.0.0`

**Comment by JeremyWuuuuu** [comment]
duplicate with #614

**Comment by lighter217** [comment]
i'm using `vue 3.0.0` and `element-plus 1.0.1-alpha.15`
![zzzz](https://user-images.githubusercontent.com/44219413/99742791-30503a00-2b0f-11eb-808f-7fa036791255.png)

**Comment by ZzqiZQute** [comment]
This issue happens because there is an issue in `@vue/babel-helper-vue-transform-on` package, it just convert the first character to upper case, but like some events in `element-plus`, such as `set-picker-option`, the helper change it to `onSet-picker-option`, but `vue3` will only recognize `onSetPickerOption`, so some events does not trigger when you use the build version.

**Comment by JeremyWuuuuu** [comment]
> This issue happens because there is an issue in `@vue/babel-helper-vue-transform-on` package, it just convert the first character to upper case, but like some events in `element-plus`, such as `set-picker-option`, the helper change it to `onSet-picker-option`, but `vue3` will only recognize `onSetPickerOption`, so some events does not trigger when you use the build version.
> 
> See element-plus/lib/index.esm.js and element-plus/lib/index.js

True, I tried it with `Dev` mode, it worked just fine, on `Prod` mode, this gets tricky.

**Comment by xiaogemm** [comment]
> This issue happens because there is an issue in `@vue/babel-helper-vue-transform-on` package, it just convert the first character to upper case, but like some events in `element-plus`, such as `set-picker-option`, the helper change it to `onSet-picker-option`, but `vue3` will only recognize `onSetPickerOption`, so some events does not trigger when you use the build version.
> 
> See element-plus/lib/index.esm.js and element-plus/lib/index.js

是的，确认就是这个问题，将index.js中的onSet-picker-option修改为onSetPickerOption后一切正常

**Comment by JeremyWuuuuu** [comment]
So this is confirmed as a bug caused by `Vue` for those of you tries to solve this bug, simply lock the `Vue` version at exact `3.0.0`, this issue will be gone, until then, we have no idea how the API would go. From my understanding, since they have already officially released the API they made should stay still instead of making drastic changes like this, as a side effect they broke our functionality.

**Comment by JeremyWuuuuu** [comment]
Linking this to https://github.com/vuejs/vue-next/issues/2540

**Comment by xiaogemm** [comment]
> So this is confirmed as a bug caused by `Vue` for those of you tries to solve this bug, simply lock the `Vue` version at exact `3.0.0`, this issue will be gone, until then, we have no idea how the API would go. From my understanding, since they have already officially released the API they made should stay still instead of making drastic changes like this, as a side effect they broke our functionality.

这就很奇怪了。我之前 Vue版本是3.0.0。是有这个问题的。换成3.0.2。问题同样存在
现在只能先用着了。同步更新版本（Vue和element+），要是有问题就先修改。直到不需要修改就可以使用。好在这个改动不太大

> Linking this to [vuejs/vue-next#2540](https://github.com/vuejs/vue-next/issues/2540)

这个就更奇怪了。codesandbo中vue版本3.0.0和3.0.2表现是不一致的，但我本地两个版本下表现一致。当然，我测试的只是组件内部默认行为：`<el-date-picker
      v-model="value1"
      type="date"
      format='YYYY-MM-DD'
      placeholder="选择日期">
    </el-date-picker>`
另外：DatePicker返回值只能是Date对象，之前的valueFormat弃用了吗？

**Comment by zazzaz** [comment]
> 另外：DatePicker返回值只能是Date对象，之前的valueFormat弃用了吗？

Yes, cause Date object is the best way to store a date time. you can check the ☝️ Breaking Change List  here https://github.com/element-plus/element-plus/issues/162#issuecomment-676052160

Besides, you can use `dayjs(date-value).format('YYYY')` to get what you want easily. And since `dayjs` is the core dependence of Element Plus, there's no additional cost.

**Comment by JeremyWuuuuu** [comment]
Should be fixed in #729 , upgrade to `Vue@^3.0.4` and `ElementPlus@^1.0.0-beta.4`. You shall have no problem ideally
